app/controllers/coronavirus_form/address_lookup_controller.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::AddressLookupController < ApplicationController
+  def submit
+    redirect_to check_your_answers_url if session[:check_answers_seen]
+
+    if params[:postcode].blank?
+      flash.now[:validation] = [{ text: I18n.t("coronavirus_form.questions.address_lookup.hint"), field: "postcode" }]
+      render controller_path, status: :unprocessable_entity
+    elsif params[:address_string].present?
+      update_params
+      redirect_to support_address_path
+    else
+      begin
+        results = helpers.postcode_lookup(params[:postcode])
```

Comment:
We should probably validate the postcode here using our postcode validator. No point making an API call with a bad postcode.

Reply:
Done.